django_mongodb/indexes.py

```diff
@@ -0,0 +1,20 @@
+from django.db.models import Index
+from django.db.models.sql.query import Query
+
+
+def _get_condition_mql(self, model, schema_editor):
+    """Analogous to Index._get_condition_sql()."""
+    query = Query(model=model, alias_cols=False)
+    where = query.build_where(self.condition)
+    compiler = query.get_compiler(connection=schema_editor.connection)
+    mql_ = where.as_mql(compiler, schema_editor.connection)
+    # Transform aggregate() query syntax into find() syntax.
```

Comment:
This stinks and doesn't work for multiple conditions / null conditions (but maybe could be done). I wonder if it would be better to have a separate path from `as_mql()` to generate `find()` compatible MQL rather than trying to convert from the aggregate syntax.

Reply:
As I understand it, we're pulling the index conditions from a model's field definitions using the compiler built by the schema editor? Is this schema_editor compiler different from our Mongo one?

can self.condition ever be a list of items?


Speaking more broadly, I'd say, yes, let's include the `find` compatible conversions, but how many nodes/expressions would we need that for? Also, since `find` is much more limited there would still be gaps.

Reply:
`query.get_compiler()` is the same compiler we use for other queries. 

`condition` [is a `Q` object](https://docs.djangoproject.com/en/dev/ref/models/indexes/#condition): "For example, `condition=Q(pages__gt=400)` indexes records with more than 400 pages."

What doesn't currently work is multiple conditions like:
```python
Q(
    pub_date__gt=datetime.datetime(
        year=2015,
        month=1,
        day=1,
        tzinfo=timezone.get_current_timezone(),
    )
)
& Q(headline__contains="China")
```
Maybe that can be translated without too much difficulty. I didn't think about it too much.

Obviously, we can only support conditions that MongoDB supports.

Reply:
🤔  we failed in the early stages when we used find instead of expr. 🤔 but maybe the indexes are different. It's not trivial to decide.

Reply:
See https://www.mongodb.com/docs/manual/core/index-partial/#create-a-partial-index. It looks like it could be a subset of $find, so maybe implementing a separate code path is feasible and best.

Reply:
🤔  I think it might be enough to re-implement the built-in lookup to use `find` instead of `expr`. This could be done with another resolver or a flag parameter. Lookups sometimes require extra parameters (not the built-in ones, but they exist); the downside is that those flags must be passed through the `where` node or injected when an expression is created. 

The `whereNode` should be adapted or a new one created, as `not`, `or`, and `and` are handled very differently.

Reply:
That's what I was thinking. Something like how we had `as_mql` and `as_mql_agg`. I leave it to you.